asm-to-pil/src/vm_to_constrained.rs

```diff
@@ -454,6 +456,86 @@ impl<T: FieldElement> ASMPILConverter<T> {
         res
     }
 
+    fn handle_instr_declaration_params(&mut self, m: &mut Machine<T>) {
+        for l in &mut m.links {
+            let lhs = &l.params;
+            let rhs = &mut l.to.params;
+            // lhs params must all be assignment registers when mapping instruction to operation
+            lhs.inputs.iter().chain(lhs.outputs.iter()).for_each(|p| {
+                assert!(p.ty.is_none());
+                assert!(self
+                    .registers
+                    .get(&p.name)
+                    .is_some_and(|r| r.ty.is_assignment()));
+            });
+
+            // if rhs is not empty, check it's valid
+            if !rhs.is_empty() {
```

Comment:
Isn't this redundant if you are iterating over empty lists otherwise?

Reply:
This also checks that all left side (assignment) registers are used on the right side.
If right side is empty (syntactic sugar for `lhs = rhs`), this check is not made.